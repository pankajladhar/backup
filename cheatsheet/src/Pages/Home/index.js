import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Home.css';

class Home extends Component {
    render() {
        return (
            <div className="Home">
                <section className="Home__LeftPanel">
                    <div className="Home__LeftPanel__Content">
                        <h1 className="Home__LeftPanel__Title">
                            <span className="sr-only">Cheatsheet</span>
                        </h1>
                        <div className="Home__LeftPanel__Content__SocialIcons">
                            <a href="https://github.com/pankajladhar/cheatsheet"
                                rel="noopener noreferrer"
                                className="Home__LeftPanel__Content__SocialIcons__GitHub"
                                target="_blank">
                                <span className="sr-only"> Github | Pankaj Ladhar </span>
                            </a>
                            <a href="https://www.linkedin.com/in/pankajladhar/"
                                rel="noopener noreferrer"
                                className="Home__LeftPanel__Content__SocialIcons__LinkedIn"
                                target="_blank">
                                <span className="sr-only"> LinkedIn | Pankaj Ladhar </span>
                            </a>
                        </div>
                    </div>
                </section>
                <section className="Home__RightPanel">
                    <div className="Home__RightPanel__Content">
                        <h2 className="Home__RightPanel__Title">Collection of Cheatsheet:</h2>
                        <ul>
                            <li>
                                <Link className="btn" to="/cheatsheet/markdown">Markdown</Link>
                            </li>
                        </ul>
                    </div>
                </section>
            </div>
        );
    }
}

export default Home;