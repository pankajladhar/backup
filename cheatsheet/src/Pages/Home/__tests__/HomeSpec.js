import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter} from 'react-router-dom'
import Home from './../index';

describe('Home', () => {
    it('renders correctly', () => {
        let tree = renderer.create(
            <StaticRouter location="someLocation" context={{}}>
                <Home  />
            </StaticRouter>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});