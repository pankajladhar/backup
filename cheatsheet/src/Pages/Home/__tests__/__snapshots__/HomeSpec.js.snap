// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="Home"
>
  <section
    className="Home__LeftPanel"
  >
    <div
      className="Home__LeftPanel__Content"
    >
      <h1
        className="Home__LeftPanel__Title"
      >
        <span
          className="sr-only"
        >
          Cheatsheet
        </span>
      </h1>
      <div
        className="Home__LeftPanel__Content__SocialIcons"
      >
        <a
          className="Home__LeftPanel__Content__SocialIcons__GitHub"
          href="https://github.com/pankajladhar/cheatsheet"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="sr-only"
          >
             Github | Pankaj Ladhar 
          </span>
        </a>
        <a
          className="Home__LeftPanel__Content__SocialIcons__LinkedIn"
          href="https://www.linkedin.com/in/pankajladhar/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="sr-only"
          >
             LinkedIn | Pankaj Ladhar 
          </span>
        </a>
      </div>
    </div>
  </section>
  <section
    className="Home__RightPanel"
  >
    <div
      className="Home__RightPanel__Content"
    >
      <h2
        className="Home__RightPanel__Title"
      >
        Collection of Cheatsheet:
      </h2>
      <ul>
        <li>
          <a
            className="btn"
            href="/cheatsheet/markdown"
            onClick={[Function]}
          >
            Markdown
          </a>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
